### PR TITLE
WIP Android bpftrace

### PR DESCRIPTION
**Pull request by dalehamel**
This build adds cross-compilation support to bpftrace, specifically targeting the Android platform, using the current latest android NDK, based on LLVM 8. 

This PR can also pave the way for general cross-compilation as it provides a means of satisfying all build and runtime dependencies of bpftrace:

- bcc pulls in libelf, and can leverage the existing embedded_llvm and embedded_clang
- libelf pulls in gnulib as a build-time dependency
- btf support pulls in bfd and opcodes (i think?)

This extends the concept of "static+libc" further to provide essentially a "static+bionic_libc", as bionic is the name of Android's libc.

The full sysroot of all dependencies is built in order to provide a bpftrace binary, and `bpftrace_test`, which can be run on an android system to check that the test suite passes there.

# Additional todo

@michalgr set up a test environment for these and was able to run bpftrace_test on an android device, these are his comments:

- On Android /tmp is not a thing, /data/local/tmp is a directory that can be used instead. I remember Joel fixing problems like that in bcc by checking $TMPDIR which on Android is set to /data/local/tmp.
- there are x86 specific register names in some tests (kprobe:f { reg(\"ip\") })
- 'sh' binary on Android is reachable from multiple locations on PATH, making all the 'uprobe:sh' tests fail.
- If I fix $PATH, change tmp dir location and run bpftrace_test on x86_64 emulator then all the tests pass.

For follow-up, it may be worth investigating in https://github.com/actions/virtual-environments/issues/183 to be able to have a means of automatically running these android tests, but that would require nested KVM which i don't think github actions supports (but maybe travis?)

### CODE REVIEW

**Comment by dalehamel** [comment]
@michalgr I rebased this and it's finally green FYI. I think it could use a little more refactoring yet, but wanted to give you a heads-up this is still progressing :)

File: cmake/embed/embed_clang.cmake [hunk 1]
```diff
@@ -82,7 +52,7 @@ endif()
 set(CLANG_CONFIGURE_FLAGS
     -Wno-dev
     -DLLVM_TARGETS_TO_BUILD=BPF
-    -DCMAKE_BUILD_TYPE=${EMBEDDED_BUILD_TYPE}
+    -DCMAKE_BUILD_TYPE=MinSizeRel
```
Comment (dalehamel) [inline cmake/embed/embed_clang.cmake]: I don't think there is any reason to build RelWithDebinfo for embedded, it bloats the size on Github Actions a lot, and only 14GB disk is available.

**Comment by challchampion** [comment]
can this run on Android without root ?

**Comment by dalehamel** [comment]
@challchampion no you need root access to do the BPF syscall and other things

**Pull request closed without merging**